Presentation.Web/app/components/it-contract/it-contract-overview.controller.ts

```diff
@@ -261,16 +269,25 @@
                                         "AssociatedSystemUsages($expand=ItSystemUsage($select=Id;$expand=ItSystem($select=Name,Disabled)))," +
                                         "DataProcessingRegistrations($select=IsAgreementConcluded)," +
                                         "LastChangedByUser($select=Name,LastName)";
-                                // if orgunit is set then the org unit filter is active
+
                                 var orgUnitId = self.$window.sessionStorage.getItem(self.orgUnitStorageKey);
+                                var query;
+                                // if orgunit is set then the org unit filter is active
                                 if (orgUnitId === null) {
-                                    return `/odata/Organizations(${self.user.currentOrganizationId})/ItContracts` +
-                                        urlParameters;
+                                    query = `/odata/Organizations(${self.user.currentOrganizationId})/ItContracts`;
                                 } else {
-                                    return `/odata/Organizations(${self.user
-                                        .currentOrganizationId})/OrganizationUnits(${orgUnitId})/ItContracts` +
-                                        urlParameters;
+                                    query = `/odata/Organizations(${self.user
+                                        .currentOrganizationId})/OrganizationUnits(${orgUnitId})/ItContracts`;
                                 }
+
+                                var criticalityId = self.$window.sessionStorage.getItem(self.criticalityTypeStorageKey);
+                                if (criticalityId === null) {
+                                    urlParameters += ",CriticalityType($select=Id)";
+                                } else {
+                                    urlParameters += `,CriticalityType($select=Id)&%24filter=CriticalityType/Id eq ${criticalityId}`;
```

Comment:
this is not how we do it.. look in the other controllers
expansion criteria work regardless of the filter..

also filter must be applied using the filter query and mutations to that. Remember that last time we went over the same problem